recommence/Config.py

```diff
@@ -19,3 +21,35 @@ def get_staging_path(self) -> str:
 
     def get_staging_data_path(self) -> str:
         return f'{self.get_staging_path()}/{self.data_file}'
+
+@dataclass
+class ReporterConfig:
+    '''
+    config type can be one of the following:
+    no_report: No report
+    logger:  Through the logger
+    file: To user specified file
+    sql: To user specified sql database location
+    '''
+    type: str
+    logger: logging.Logger
+    file_save_path: str | None = None
+    database_path: str | None = None
+
+    def should_report(self) -> bool:
+        return self.type != 'no_report'
+
+    def get_report_path(self) -> Any:
+        if self.type == 'file' and self.file_save_path is not None:
+            if not os.path.exists(os.path.dirname(self.file_save_path)):
+                os.makedirs(os.path.dirname(self.file_save_path))
```

Comment:
We should avoid side-effects in a `getter` method. One way around this might be to go ahead and formalize these various writer backends into their own classes:
```python
class ReporterBackend:
  @abstractmethod
  def write(self, report: Report) -> None: ...

  @abstractmethod
  def prep(self) -> None: ...


class FileReporterBackend(ReporterBackend):
  def prep(self):
    os.makedirs(file_dir, exists_ok=True)
  
  def write(self, report: Report):
    with open(report_path, 'w') as f:
      f.write(...)
```

Then this prep work of ensuring the save directory is built can be done in a `.prep()` lifecycle method.